src/events/messageCreateHandler.ts

```diff
@@ -0,0 +1,298 @@
+        }
+        if (isEmbed) {
+            const commandEmbedData = commandEmbed(commandTitle, actualCommandContent, embedColor, commandImage);
+            const { footer } = message.embeds[0];
+            const newFooter = footer?.text ? `${footer.text} - The choice has expired.` : 'The choice has expired.';
+            commandEmbedData.setFooter({ text: newFooter });
+            return message.edit({ embeds: [commandEmbedData], components: [] });
+        }
+
+        const content: string[] = [];
+        if (commandTitle) {
+            content.push(`**${commandTitle}**`);
+        }
+        content.push(actualCommandContent);
+        content.push('\n`The choice has expired.`');
+        return message.edit({
+            content: makeLines(content),
+            components: [],
+        });
+    } catch (error) {
+        Logger.error(error);
+        return message.reply('An error occurred while updating the message.');
+    }
+}
+
+async function sendPermError(message: Message, errorText: string) {
+    if (constantsConfig.prefixCommandPermissionDelay > 0) {
+        errorText += `\n\nThis message & the original command message will be deleted in ${constantsConfig.prefixCommandPermissionDelay / 1000} seconds.`;
+    }
+    const permReply = await sendReply(message, 'Permission Error', errorText, true, constantsConfig.colors.FBW_RED, '');
+    if (constantsConfig.prefixCommandPermissionDelay > 0) {
+        setTimeout(() => {
+            try {
+                permReply.delete();
+                message.delete();
+            } catch (error) {
+                Logger.error(`Error while deleting permission error message for command: ${error}`);
+            }
+        }, constantsConfig.prefixCommandPermissionDelay);
+    }
+}
+
+export default event(Events.MessageCreate, async (_, message) => {
+    const { id: messageId, author, channel, content } = message;
+    const { id: authorId, bot } = author;
+
+    if (bot || channel.isDMBased()) return;
+    const { id: channelId, guild } = channel;
+    const { id: guildId } = guild;
+    Logger.debug(`Processing message ${messageId} from user ${authorId} in channel ${channelId} of server ${guildId}.`);
+
+    const inMemoryCache = getInMemoryCache();
+    if (inMemoryCache && content.startsWith(constantsConfig.prefixCommandPrefix)) {
+        const commandTextMatch = content.match(`^\\${constantsConfig.prefixCommandPrefix}([\\w\\d-_]+)[^\\w\\d-_]*([\\w\\d-_]+)?`);
+        if (commandTextMatch) {
+            let [commandText] = commandTextMatch.slice(1);
+            const commandVersionExplicitGeneric = (commandText.toLowerCase() === 'generic');
+
+            // Step 1: Check if the command is actually a version alias
+            const commandCachedVersion = await inMemoryCache.get(`${memoryCachePrefixVersion}:${commandText.toLowerCase()}`);
+            let commandVersionId;
+            let commandVersionName;
+            let commandVersionEnabled;
```

Comment:
```suggestion
            let commandVersionId: string;
            let commandVersionName: string;
            let commandVersionEnabled: boolean;
```
I think this should be possible? I would prefer this as it gets rid of `any`.